### PR TITLE
Add an example for adding HDF5 logging to a graph

### PR DESCRIPTION
**Pull request by pdamodara**
## Description
The process for enabling HDF5 logging involves a few steps.
- This PR adds an example for reference.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Feature/Issue validation/testing
- [x] Run the example graph `python -m labgraph.examples.graph_w_logging`
- Verified that a HDF5 file named `graph_w_logging.h5` is created in the current directory.

## Checklist:
- Have you added tests that prove your fix is effective or that this feature works?
   - N/A, this is an example only and will be run to understand behavior; No unit tests required
- [x] New and existing unit tests pass locally with these changes?
- [x] Has code been commented, particularly in hard-to-understand areas?
- Have you made corresponding changes to the documentation?
   - N/A, this example is intended to be documentation

### CODE REVIEW

File: labgraph/examples/graph_w_logging.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    GENERATOR: RandomFloatGenerator
+
+    def process_modules(self) -> Tuple[lg.Module, ...]:
+        return (self.GENERATOR,)
+
+    def logging(self) -> Dict[str, lg.Topic]:
+        return {"random_data": self.GENERATOR.OUTPUT}
+
+
```
Comment (jfResearchEng) [inline labgraph/examples/graph_w_logging.py:44]: We can add in the following comments on the expected behavior of the function:
graph_w_logging.hdf5 will be created under folder output_directory

File: labgraph/examples/graph_w_logging.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python3
```
Comment (jfResearchEng) [inline labgraph/examples/graph_w_logging.py:1]: In the future or now, I can re-structure the example folder. For example, this example can be put in a logging folder.

**Review by jfResearchEng — APPROVED** [review]
Thanks for adding this useful HDF5 logging example!!

File: labgraph/examples/graph_w_logging.py (hunk 1 above)
Comment (pdamodara) [reply labgraph/examples/graph_w_logging.py:44]: The docstring for the class mentions the expected behavior, i.e., logging to the current directory - Would you prefer that I move that docstring to this function?

File: labgraph/examples/graph_w_logging.py (hunk 2 above)
Comment (pdamodara) [reply labgraph/examples/graph_w_logging.py:1]: +1, I think that's a great idea.

**Pull request merged**